### PR TITLE
sovereign extra header verifier in runtype components

### PR DESCRIPTION
**Pull request by axenteoctavian**
## Reasoning behind the pull request
- Sovereign extra header verifier in runtype components
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: factory/runType/sovereignRunTypeComponents.go [hunk 1]
```diff
@@ -202,7 +196,11 @@ func (rcf *sovereignRunTypeComponentsFactory) Create() (*runTypeComponents, erro
 
 	expiryTime := time.Second * time.Duration(rcf.sovConfig.OutgoingSubscribedEvents.TimeToWaitForUnconfirmedOutGoingOperationInSeconds)
 
-	err = rtc.extraHeaderSigVerifierHandler.RegisterExtraHeaderSigVerifier(rcf.extraVerifier)
+	sovHeaderSigVerifier, err := headerCheck.NewSovereignHeaderSigVerifier(rcf.cryptoComponents.BlockSigner())
```
Comment (mariusmihaic) [inline factory/runType/sovereignRunTypeComponents.go:199]: Is this also removed from normal run type ?

Comment (axenteoctavian) [reply factory/runType/sovereignRunTypeComponents.go:199]: Normal RunType doesn't use an extra header sig verifier, it's just:
```extraHeaderSigVerifierHandler:           headerCheck.NewExtraHeaderSigVerifierHolder(),```

Comment (mariusmihaic) [reply factory/runType/sovereignRunTypeComponents.go:209]: The name of the variable was wrong and confused me;
extraHeaderSigVerifierHandler -> this is a holder, not a handler, please rename it

**Review by mariusmihaic — CHANGES_REQUESTED** [review]

File: factory/runType/runTypeComponentsHandler.go [hunk 2]
```diff
@@ -544,16 +544,16 @@ func (mrc *managedRunTypeComponents) TxPreProcessorCreator() preprocess.TxPrePro
 	return mrc.runTypeComponents.txPreProcessorCreator
 }
 
-// ExtraHeaderSigVerifierHandler returns the extra header sig verifier handler
-func (mrc *managedRunTypeComponents) ExtraHeaderSigVerifierHandler() headerCheck.ExtraHeaderSigVerifierHolder {
+// ExtraHeaderSigVerifierHolder returns the extra header sig verifier handler
```
Comment (mariusmihaic) [inline factory/runType/runTypeComponentsHandler.go:547]:
```suggestion
// ExtraHeaderSigVerifierHolder returns the extra header sig verifier holder
```

**Review by mariusmihaic — CHANGES_REQUESTED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**